### PR TITLE
Feature/view sch details

### PR DESCRIPTION
**Pull request by TimOsse**

### CODE REVIEW

File: frontend/src/components/school/SchoolDetails.vue [hunk 1]
```diff
@@ -0,0 +1,501 @@
+<template>
+  <v-container class="containerSetup" fluid>
+    <div style="width: 100%;" :overlay=false>
+      <v-row class="pt-0"
+             :class="{'mr-0 ml-0': $vuetify.breakpoint.smAndDown, 'mr-3 ml-3': $vuetify.breakpoint.mdAndUp}">
+        <v-col cols="10 pt-0">
+          <div>
+          <!--  <v-row> -->
+              <!--<v-col class="pb-0 pt-0 d-flex justify-start"> -->
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:9]: Commented code. 

Cols should be within a row ex. <row> <col> 

if you face padding issues try no-gutters in row. ex. <row no-gutters>

File: frontend/src/components/school/SchoolDetails.vue [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+                    <h2 class="subjectHeading">{{school.mincode}} - {{school.displayName}}</h2>
+                    <div class="ministryOwnershipTeamName" style="color: black">{{district.districtNumber}} - {{district.name}}</div>
+                  </v-col>
+                  <v-col cols="12" lg="2" class="d-flex justify-end">
+                    <PrimaryButton width="8em" primary icon="mdi-pencil" text="Edit"></PrimaryButton>
+                  </v-col>
+                </v-row>
+                <v-row>
+                  <v-col cols="2" lg="1" class="pb-0 pt-0">
+                    <v-row class="mb-n4">
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:22]: I think you can probably get rid of the mb-n4 if you use the no-gutters trick.

File: frontend/src/components/school/SchoolDetails.vue [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+                        <span class="ml-n1">{{ formatPhoneNumber(school.faxNumber) }}</span>
+                      </v-col>
+                    </v-row>
+                  </v-col>
+                  <v-col cols="4" lg="2" class="pb-0 pt-0">
+                    <v-row class="mb-n4">
+                      <v-col cols="10" class="pb-2 pt-0 pr-0 pl-10">
+                         <v-btn icon :href="`${school.website}`" target="_blank">
+                          <v-icon aria-hidden="false" class="pr-3">mdi-web</v-icon>
+                           {{school.website}} https://www.google.com
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:66]: google.com?

File: frontend/src/components/school/SchoolDetails.vue [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+    getThisSchoolsDetails(){
+      this.school = '';
+
+      ApiService.apiAxios.get(ApiRoutes.school.SCHOOL_DETAILS_BY_ID + `/${this.userInfo.activeInstituteIdentifier}`)
+        .then(response => {
+          this.school = response.data;
+          this.populateExtraSchoolFields(this.school);
+          this.getDistrictDetails(this.school.districtId);
+        }).catch(error => {
+        //to do add the alert framework for error or success
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:279]: Ahh there's an existing alert framework for this. https://github.com/bcgov/EDUC-EDX/blob/master/frontend/src/mixins/alertMixin.js 

take a peek at this for implementation https://github.com/bcgov/EDUC-EDX/blob/master/frontend/src/components/SecureExchange/MessageDisplay.vue#:~:text=import%20alertMixin%20from%20%27%40/mixins/alertMixin%27%3B

File: frontend/src/components/school/SchoolDetails.vue [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+          this.loadingTable = false;
+        });
+    },
+    getDistrictDetails(districtId){
+      this.district = '';
+      ApiService.apiAxios.get(ApiRoutes.institute.DISTRICT + '/'+ districtId)
+        .then(response => {
+          this.district = response.data;
+        }).catch(error => {
+        //to do add the alert framework for error or success
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:291]: same as above.

File: frontend/src/components/school/SchoolDetails.vue [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+          gradeList.push('11');
+          break;
+        case 'GRADE12':
+          gradeList.push('12');
+          break;
+        }
+      }
+      gradeList.sort();
+      return gradeList.toString();
+    },
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:354]: I think we have these gradeCodes from the student-api (more info in chat)

File: frontend/src/components/school/SchoolDetails.vue [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+      }
+    },
+    formatDate(date){
+      if(date !== '' && date !== null) {
+        return new Date(date).toISOString().slice(0, 10).replace(/-/g, '/');
+      } else {
+        return '-';
+      }
+    },
+    formatPhoneNumber(phNumber){
```
Comment (SoLetsDev) [inline frontend/src/components/school/SchoolDetails.vue:476]: There's a formatPhoneNumber function in the shared utils. https://github.com/bcgov/EDUC-STUDENT-ADMIN/blob/master/frontend/src/utils/format.js#:~:text=%7D-,export%20function%20formatPhoneNumber(phoneNumber)%20%7B,-if%20(!

Would suggest that we can put the formatDate function in this file too

Comment (TimOsse) [reply frontend/src/components/school/SchoolDetails.vue:476]: That formatDate function returns a date that looks like the original including the time that we don't want.

**Review by arcshiftsolutions — APPROVED** [review]

**Pull request merged**